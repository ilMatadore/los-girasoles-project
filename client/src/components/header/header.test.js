// import React from 'react';
// import { shallow, mount, render } from 'enzyme';
// import Header from './header.component';
// import { UserContext } from '../../context/userContext/user.context';
// import { Router, MemoryRouter } from "react-router-dom";

// jest.mock('react-router-dom', () => ({
//     useLocation: jest.fn().mockReturnValue({
//       pathname: 'localhost:3000/',
//     }),
// }));

// describe('render Header', () => {

//     // it("should render Header", () => {
//     //     const wrapper = render(
//     //         <MemoryRouter>
//     //             <Header/>
//     //         </MemoryRouter>
//     //     )
//     //     // const wrapper = mount(<Header />);
//     //     expect(wrapper.find(".HeaderContainer").exists()).toEqual(true)
//     // })


//     it(">>>> should render user first name in header", () => {

//         const wrapper = mount(
//             <UserContext.Provider
//               value={{
//                 id: '1',
//                 first_name: 'Test',
//                 last_name: 'Tester',
//                 email: 'test@test.com',
//                 address: '123 test',
//                 phone: '123124',
//                 city: 'test city',
//                 state: 'ts',
//                 successLogin: jest.fn(),
//                 userLogout: jest.fn(),
//                 updateProfile: jest.fn()
//               }}
//             >
//                 <MemoryRouter>
//                     <Header />
//                 </MemoryRouter>
//             </UserContext.Provider>
//           );
//           expect(wrapper.find(".header-user-name").text()).toEqual("Hola Test");
            
//     })
    
// });


import React from 'react';
import { shallow, render } from 'enzyme';
import Header from './header.component';
import { UserContext } from '../../context/userContext/user.context';


describe('Header Tests', () => {

    const wrapper = shallow(<Header />)

    it('Expect to render Header component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('To Not display user first name', () => {        
        expect(wrapper.find(".header-user-name")).toHaveLength(0)
    });

    // it('displays user first name in header', () => {
    //     const view = render(<UserContext.Provider value={{ first_name: "Tester"}}><Header /></UserContext.Provider>)
    //     // expect(view.find(".header-user-name")).toHaveLength(1)
    //     // console.log(view[0].children[1])
    // })




})


