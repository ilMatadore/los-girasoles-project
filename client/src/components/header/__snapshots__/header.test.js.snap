// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Tests Expect to render Header component 1`] = `
<div
  className="HeaderContainer"
>
  <div
    className="LogoContainer"
  >
    <img
      alt="logo"
      className="logo"
      src="logo3.png"
    />
    <span
      className="title"
    >
      Los Girasoles
    </span>
  </div>
  <div
    className="OptionsContainer"
  >
    <div
      className="OptionLink"
      id="Menu"
      onClick={[Function]}
      to="/"
    >
      <FaBars
        style={
          Object {
            "fontSize": "30px",
          }
        }
      />
    </div>
    <Link
      className="OptionLink"
      id="Home"
      to="/"
    >
      <FaHome
        style={
          Object {
            "fontSize": "30px",
          }
        }
      />
    </Link>
    <Link
      className="OptionLink"
      id="Quienes"
      to="/quienes"
    >
      <FaAddressCard
        style={
          Object {
            "fontSize": "30px",
          }
        }
      />
    </Link>
    <Link
      className="OptionLink"
      id="Contact"
      to="/contact"
    >
      <FaEnvelope
        style={
          Object {
            "fontSize": "30px",
          }
        }
      />
    </Link>
    <Link
      className="OptionLink"
      id="Cart"
      to="/cart"
    >
      <FaShoppingCart
        style={
          Object {
            "fontSize": "30px",
          }
        }
      />
    </Link>
    <Link
      className="OptionLink"
      id="SignIn"
      to="/signin"
    >
      <FaSignInAlt
        style={
          Object {
            "fontSize": "30px",
          }
        }
      />
    </Link>
  </div>
</div>
`;
